server/api/profiles.js

```diff
@@ -50,14 +50,13 @@ router.get("/:profileId", (req, res) => {
 		});
 });
 //checks if email is not unique
-router.get("/:email", (req, res) => {
+router.get("/email/:email", (req, res) => {
 
 	const email = req.params.email;
 	usersDb
 		.getProfileByEmail(email)
 		.then((data) => {
 			res.send(data);
```

Comment:
You should not send data if the query result is empty or undefined. Add a check on data and send an HTTP 404 if it's empty, like this:
```
if(data && data !== undefined)
	res.send(data);
else res.sendStatus(404);
```